test/format/format.sh

```diff
@@ -0,0 +1,301 @@
+}
+[[ -d "$home" ]] || {
+	echo "$0: run directory \"$home\" not found"
+	exit 1
+}
+
+[[ $verbose -ne 0 ]] &&
+    echo "$0 [-c $config] [-h $home] [-j $jobs] [-n $runs] [-t $minutes] $format_args"
+
+failure=0
+success=0
+# Resolve/cleanup completed jobs.
+resolve()
+{
+	for dir in $home/RUNDIR.[0-9]*; do
+		# Skip log files and failures we've already reported.
+		[[ ! -d $dir ]] || [[ -f "$dir/reported" ]] && continue
+		log="$dir.log"
+
+		# Discard successful runs.
+		grep 'successful run completed' $log > /dev/null && {
+			rm -rf $dir $log
+			success=$(($success + 1))
+			[[ $verbose -ne 0 ]] && echo "$0: job in $dir successfully completed"
+			continue
+		}
+
+		# Test recovery on runs configured for random abort. */
+		grep 'aborting to test recovery' $log > /dev/null && {
+			cp -pr $dir $dir.RECOVER
+
+			(echo
+			 echo "$0: running recovery after abort test"
+			 echo "$0: original directory copied into $dir.RECOVER"
+			 echo) >> $log
+
+			# Everything is a table unless explicitly a file.
+			uri="table:wt"
+			grep 'data_source=file' $dir/CONFIG > /dev/null && uri="file:wt"
+						
+			# Use the wt utility to recover & verify the object.
+			if  $($wt_binary -R -h $dir verify $uri >> $log 2>&1); then
+				rm -rf $dir $dir.RECOVER $log
+				success=$(($success + 1))
+				[[ $verbose -ne 0 ]] &&
+				    echo "$0: job in $dir successfully completed"
+			else
+				echo "$0: failure status reported" > $dir/reported
+				failure=$(($failure + 1))
+				echo "$0: job in $dir failed abort/recovery testing"
+			fi
+			continue
+		}
+
+		# Discard runs where the timer went off.
+		grep 'caught signal' $log > /dev/null && {
+			rm -rf $dir $log
+			[[ $verbose -ne 0 ]] && echo "$0: job in $dir aborted"
+			continue
+		}
+
+		# Report failures.
+		grep 'RUN PARAMETERS' $log > /dev/null && {
+			echo "$0: failure status reported" > $dir/reported
+			failure=$(($failure + 1))
+			echo "$0: job in $dir failed"
+		}
+	done
+	return 0
+}
+
+# Start a single job.
+count_runs=0
+run_format()
+{
+	count_runs=$(($count_runs + 1))
+	dir="$home/RUNDIR.$count_runs"
+	log="$dir.log"
+
+	if [[ $smoke_test -ne 0 ]]; then
+		args=${smoke_list[$smoke_next]}
+		smoke_next=$(($smoke_next + 1))
+		echo "$0: starting smoke-test job in $dir"
+	else
+		args=$format_args
+
+		# If abort/recovery testing is configured, do it 5% of the time.
+		[[ $abort_test -ne 0 ]] && [[ $(($count_runs % 20)) -eq 0 ]] && args="$args abort=1"
+
+		echo "$0: starting job in $dir"
+	fi
+	[[ $verbose -ne 0 ]] &&
+	    echo "$0: running $format_binary -c "$config" -h "$dir" -1 $args quiet=1"
+	$format_binary -c "$config" -h "$dir" -1 $args quiet=1 > $log 2>&1 &
+}
+
+seconds=$((minutes * 60))
+start_time="$(date -u +%s)"
+while :; do
+	# Check if our time has expired.
+	[[ $seconds -ne 0 ]] && {
+		now="$(date -u +%s)"
+		elapsed=$(($now - $start_time))
+
+		# If we've run out of time, terminate all running jobs.
+		[[ $elapsed -ge $seconds ]] && {
+			stop=1
+			forcequit=1
+		}
+	}
```

Comment:
As this code block locates at the beginning of the outer while loop, the whole test could run longer compared to the configured value `-t <mins>`, as the inner while loop could take time (depending on the `-n <runs>` and `-j <jobs>` settings, and there's an additional 10 sec sleep at the end of the outer while loop.

Reply:
Yes, you are correct. I didn't try to make format run for a specific number of seconds, it will quite likely take some number of seconds to exit.

Can you discuss why that's a problem?

Reply:
I don't think it's a problem - it's not a big deal to have some additional sleep/exit time compared to the configured minutes to run. Just hope to line up expectation for the `-t <mins>` setting.